### PR TITLE
Add Node Intro + Express Workshop

### PR DESCRIPTION
**Pull request by cedric-h**
I'm sure there are still some typos in there somewhere ;o;

I can't wait to write the next one about WebSockets! :D

### CODE REVIEW

File: workshops/multiplayer/README.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+A webserver is a little bit of software that lets you send websites -- and all of the files that they need -- to anyone who connects to your server.
+I'll admit, making a webserver seems a little prosaic.
+It's not even immediately evident *why* you'd want to bother making one.
+In truth, I'm making this workshop as the beginning of a tutorial series I'm running on how to make a webgame with networked multiplayer, but there really are plenty of other reasons to make your own webserver.
+
+First of all, I think you should do this workshop just to get more familiar with navigating through your own filesystem using the **command line**, if you aren't already.
+I introduce a few commands as I go through, and once you have these you're only a couple extra googles away from becoming fairly proficient at using the command line.
+That can make you a good bit faster when manipulating the filesystem, especially if you're already a fast typist.
+
+**Node.js** is a pretty cool technology for doing lots of things, especially developing web related software. This project is an awesome introduction to node,
```
Comment (andrewgcodes) [inline workshops/multiplayer/README.md]: No comma before because because it's dependent clause.

File: workshops/multiplayer/README.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+order: 9
+---
+
+# Express Webserver
+
+## What can you learn in this workshop?
+A webserver is a little bit of software that lets you send websites -- and all of the files that they need -- to anyone who connects to your server.
+I'll admit, making a webserver seems a little prosaic.
+It's not even immediately evident *why* you'd want to bother making one.
+In truth, I'm making this workshop as the beginning of a tutorial series I'm running on how to make a webgame with networked multiplayer, but there really are plenty of other reasons to make your own webserver.
```
Comment (andrewgcodes) [inline workshops/multiplayer/README.md]: "really" is unnecessary and a bit strange sounding (maybe cuz my socal english) ur call

File: workshops/multiplayer/README.md (hunk 1 above)
Comment (andrewgcodes) [reply workshops/multiplayer/README.md]: sorry its because the secondf half is a subordinate clause

File: workshops/multiplayer/README.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+I introduce a few commands as I go through, and once you have these you're only a couple extra googles away from becoming fairly proficient at using the command line.
+That can make you a good bit faster when manipulating the filesystem, especially if you're already a fast typist.
+
+**Node.js** is a pretty cool technology for doing lots of things, especially developing web related software. This project is an awesome introduction to node,
+because it walks you through configuring a brand new node project.
+Surely that could come in handy at a Hackathon some time in the future, when you want to use a library for Machine Learning or a library for some other fancy concept that uses Node.
+
+**Express** is awesome for quickly hosting files from your own computer.
+There are a bunch of other ways to host files from your computer, including some command line interfaces that don't require any code at all,
+but because Express is just Node.js code it offers a lot of opportunities for doing some fairly complex configuration and web app serving.
```
Comment (andrewgcodes) [inline workshops/multiplayer/README.md]: comma between code and it

File: workshops/multiplayer/README.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+Perhaps it's overkill for a webgame which has fairly simple webserver needs, but I like to keep my options open in case I want to add some pages to my app for making an account or looking at your player stats and whatnot.
+
+`Repl.it`/`Glitch`/`Github Pages` do offer their own interfaces you can use to get your files hosted, but here's a few reasons why you'd want to write your own with Express:
```
Comment (andrewgcodes) [inline workshops/multiplayer/README.md]: here are not here is

File: workshops/multiplayer/README.md [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+- **More configuration**. Using express, you can very carefully tweak how your files are served.
+- **Control over hosting**\*, Express isn't tied to any specific platform like the built in hosting of Glitch and Repl.it. This means that you can host your app on whatever hardware you want.
+
+\* You can still get Repl.it or Glitch (but not Github Pages, it's more limited) to host your Node/Express server for you, you don't have to use their hosting solutions when you make your project on their platforms.
```
Comment (andrewgcodes) [inline workshops/multiplayer/README.md]: split into two sentences Currently has a comma splice

File: workshops/multiplayer/README.md [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+Instead of making all of the important decisions in the application,
+the frontend just displays all of the data that the backend gives it.
+To continue *The Matrix* analogy, the frontend is like those little pods that each of the people plug themselves into,
+if those pods took the data from the matrix and turned it into a format the people's brains could understand.
+Our frontend will be made using HTML5.
+The client uses the frontend to communicate with the backend and understand the information that the backend has.
+
+## Getting Started: Making the Matrix
+We'll start with the backend. Because the backend tells the frontend what to render, it makes sense to start at the backend and work our way out.
+I'll be doing this on my local filesystem using the command line, but if you're more comfortable in repl.it feel free to follow along there.
```
Comment (andrewgcodes) [inline workshops/multiplayer/README.md]: comma between replit and feel

File: workshops/multiplayer/README.md (hunk 2 above)
Comment (cedric-h) [reply workshops/multiplayer/README.md]: interesting, definitely doesn't sound off to me but I'll go ahead and remove it.

File: workshops/node_intro/README.md [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+**Node.js** is a pretty cool technology for doing lots of things, especially developing web related software. This project is an awesome introduction to node
+because it walks you through configuring a brand new node project.
+Surely that could come in handy at a Hackathon some time in the future, when you want to use a library for Machine Learning or a library for some other fancy concept that uses Node.
+
+**Express** is awesome for quickly hosting files from your own computer.
+There are a bunch of other ways to host files from your computer, including some command line interfaces that don't require any code at all,
+but because Express is just Node.js code, it offers a lot of opportunities for doing some fairly complex configuration and web app serving.
+Perhaps it's overkill for a webgame which has fairly simple webserver needs, but I like to keep my options open in case I want to add some pages to my app for making an account or looking at your player stats and whatnot.
+
+`Repl.it`/`Glitch`/`Github Pages` do offer their own interfaces you can use to get your files hosted, but here are a few reasons why you'd want to write your own with Express:
```
Comment (harrisbegca) [inline workshops/node_intro/README.md]: Change to:
`Repl.it`, `Glitch`, and `Github Pages` do offer their own file hosting services, but here are a few reasons why you'd want to write your own with Express:

File: workshops/node_intro/README.md [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+- **More configuration**. Using express, you can very carefully tweak how your files are served.
```
Comment (harrisbegca) [inline workshops/node_intro/README.md]: Change "express" to "Express" (pretty trivial)

File: workshops/node_intro/README.md [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+Instead of making all of the important decisions in the application,
+the frontend just displays all of the data that the backend gives it.
+To continue *The Matrix* analogy, the frontend is like those little pods that each of the people plug themselves into,
+if those pods took the data from the matrix and turned it into a format the people's brains could understand.
+Our frontend will be made using HTML5.
+The client uses the frontend to communicate with the backend and understand the information that the backend has.
+
+## Getting Started: Making the Matrix
+We'll start with the backend. Because the backend tells the frontend what to render, it makes sense to start at the backend and work our way out.
+I'll be doing this on my local filesystem using the command line, but if you're more comfortable in repl.it, feel free to follow along there.
```
Comment (harrisbegca) [inline workshops/node_intro/README.md]: Change "repl.it" to "Repl.it"

File: workshops/node_intro/README.md [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+In our case, we'll want to send anyone who connects to our server the HTML file for our frontend and any images, `.js` files, `.css` files, and maybe sound files that the app's frontend might need.
+
+To do that, we're writing the little bit of the backend that sends the frontend to their web browser.  
+This terminal command will download Express for us so that we can use it in our code.  
+`npm install express --save`
+
+### Writing the Code
+Next, let's make an `index.js` file in the text editor of your choice.
+
+I use Vim. Sublime Text is alright, as are Atom, VS Code, ect.
```
Comment (harrisbegca) [inline workshops/node_intro/README.md]: ect. -> etc.

File: workshops/node_intro/README.md [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+Express has a ton of awesome content for writing more complex web apps, and I would encourage you to dive into their documentation to learn more.  
+https://expressjs.com/  
+https://nodejs.org/en/  
+
+Now you have a solid webserver you can connect to to get a webpage from.
+You could use this to run a game or a simple website.
+You now know the core concepts and terminology of the server/client architecture that the web is built on.
+If you hear someone refer to themselves as a "Frontend" or "Backend" dev, you'll know what they mean.
+(If they say "Full Stack", they mean they do both.)
+Along the way, you also learned some terminal commands and how to start a new Node.js project.
```
Comment (harrisbegca) [inline workshops/node_intro/README.md]: Might be worth noting that you can use Express in conjunction with other frameworks? (e.g. React, Django)

File: workshops/node_intro/README.md [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+
+First of all, I think you should do this workshop just to get more familiar with navigating through your own filesystem using the **command line**, if you aren't already.
+I introduce a few commands as I go through, and once you have these you're only a couple extra googles away from becoming fairly proficient at using the command line.
+That can make you a good bit faster when manipulating the filesystem, especially if you're already a fast typist.
+
+**Node.js** is a pretty cool technology for doing lots of things, especially developing web related software. This project is an awesome introduction to node
+because it walks you through configuring a brand new node project.
+Surely that could come in handy at a Hackathon some time in the future, when you want to use a library for Machine Learning or a library for some other fancy concept that uses Node.
+
+**Express** is awesome for quickly hosting files from your own computer.
```
Comment (harrisbegca) [inline workshops/node_intro/README.md]: May be hard for people new to the web dev world to understand

File: workshops/node_intro/README.md [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+There are a bunch of other ways to host files from your computer, including some command line interfaces that don't require any code at all,
+but because Express is just Node.js code, it offers a lot of opportunities for doing some fairly complex configuration and web app serving.
+Perhaps it's overkill for a webgame which has fairly simple webserver needs, but I like to keep my options open in case I want to add some pages to my app for making an account or looking at your player stats and whatnot.
+
+`Repl.it`/`Glitch`/`Github Pages` do offer their own interfaces you can use to get your files hosted, but here are a few reasons why you'd want to write your own with Express:
+- **More configuration**. Using express, you can very carefully tweak how your files are served.
+- **Control over hosting**\*, Express isn't tied to any specific platform like the built in hosting of Glitch and Repl.it. This means that you can host your app on whatever hardware you want.
+
+\* You can still get Repl.it or Glitch (but not Github Pages, it's more limited) to host your Node/Express server for you. You don't have to use their hosting solutions when you make your project on their platforms.
```
Comment (harrisbegca) [inline workshops/node_intro/README.md]: Github -> GitHub

**Comment by cedric-h** [comment]
Thanks so much for theses suggestions, Harris. I've been meaning to get to them but I've been pretty busy.

**Comment by MatthewStanciu** [comment]
Hey @cedric-h, I'm going to close this for now. This semester, we're rethinking what the Workshops page looks like and what a workshop is. I want to evaluate the current workshops and introduce some personal controlled experiments, and right now there unfortunately isn't room to consider this. Happy to reopen this and discuss more in a few months after I finish this project :)

**Pull request closed without merging**